test/AppTest.php

```diff
@@ -0,0 +1,34 @@
+<?php
+namespace ClassicAppTest;
+
+use PHPUnit_Framework_TestCase;
+use ClassicApp\EventListener\DispatcherExceptionListener;
+use GianArb\Penny\App;
+use GianArb\Penny\Config\Loader;
+use Zend\Diactoros\Request;
+use Zend\Diactoros\Response;
+use Zend\Diactoros\Uri;
+
+class AppTest extends PHPUnit_Framework_TestCase
+{
+    public function testDispatcherErrorCallsListenerIndex()
+    {
+        $spy = $this->getMockBuilder(DispatcherExceptionListener::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $spy->expects($this->once())
+            ->method('onError')
+            ->with($this->anything());
+
+        $response = $this->getMock(Response::class);
+        $request = (new Request())
+            ->withUri(new Uri('/pnf'))
+            ->withMethod('GET');
+
```

Comment:
I think it can be simplified with prophecy to make consistent with controller tests.
```php
use Prophecy\Argument;

$spy = $this->prophesize(DispatcherExceptionListener::class);
$spy->onError(Argument::any())->shouldBeCalled();

$response = $this->prophesize(Response::class);
// ..
$container->set(DispatcherExceptionListener::class, $spy->reveal());
// ...
$app->run($request, $response->reveal());
```